tests/test_utils.py

```diff
@@ -0,0 +1,136 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import unittest
+
+import lsst.utils.tests
+from lsst.daf.butler.tests import makeTestCollection, makeTestRepo
+from lsst.daf.butler.tests.utils import makeTestTempDir, removeTestTempDir
+from lsst.pipe.base import Pipeline
+from lsst.source.injection import ExposureInjectTask, ingest_injection_catalog, make_injection_pipeline
+from lsst.source.injection.utils.test_utils import (
+    make_test_exposure,
+    make_test_injection_catalog,
+    make_test_reference_pipeline,
+)
+from lsst.utils.tests import MemoryTestCase, TestCase
+
+TEST_DIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class SourceInjectionUtilsTestCase(TestCase):
+    """Test the utility functions in the source_injection package."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.root = makeTestTempDir(TEST_DIR)
+        cls.creator_butler = makeTestRepo(cls.root)
+        cls.writeable_butler = makeTestCollection(cls.creator_butler)
+
+    @classmethod
+    def tearDownClass(cls):
+        del cls.writeable_butler
+        del cls.creator_butler
+        removeTestTempDir(cls.root)
+
+    def setUp(self):
+        self.exposure = make_test_exposure()
+        self.injection_catalog = make_test_injection_catalog(
+            self.exposure.getWcs(),
+            self.exposure.getBBox(),
+        )
+        self.reference_pipeline = make_test_reference_pipeline()
+
+    def tearDown(self):
+        del self.exposure
+        del self.injection_catalog
+        del self.reference_pipeline
+
+    def test_generate_injection_catalog(self):
+        self.assertEqual(len(self.injection_catalog), 30)
+        expected_columns = {"injection_id", "ra", "dec", "source_type", "mag"}
+        self.assertEqual(set(self.injection_catalog.columns), expected_columns)
+
+    def test_make_injection_pipeline(self):
+        injection_pipeline = Pipeline("reference_pipeline")
+        injection_pipeline.addTask(ExposureInjectTask, "inject_exposure")
+        merged_pipeline = make_injection_pipeline(
+            dataset_type_name="postISRCCD",
+            reference_pipeline=self.reference_pipeline,
+            injection_pipeline=injection_pipeline,
+            exclude_subsets=False,
+            prefix="injected_",
+            instrument="lsst.obs.subaru.HyperSuprimeCam",
+            log_level=logging.DEBUG,
+        )
+        expanded_pipeline = merged_pipeline.toExpandedPipeline()
+        expected_subset_tasks = ["isr", "inject_exposure", "characterizeImage"]
+        merged_task_subsets = [merged_pipeline.findSubsetsWithLabel(x) for x in expected_subset_tasks]
+        self.assertEqual(len(merged_task_subsets), len(expected_subset_tasks))
```

Comment:
This might just be a naming thing, but is this asking that the number of produced subsets is the same as the number of expected tasks?  I woudn't have naively thought that tasks are equivalent to subsets in a way that makes sense here.  (But maybe my pipe_base knowledge is just lacking).

Reply:
The object `merged_task_subsets` is a list which gives the name of the subset which the task lives in. In this case, `merged_task_subsets` looks like this:
```
[{'test_subset'}, {'test_subset'}, {'test_subset'}]
```
i.e., the same string repeated thrice. This is just testing that all three of the tasks correctly live inside the `test_subset` subset, including the newly added `inject_exposure` task which sits between `isr` and `characterizeImage`.